Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -88,6 +107,12 @@ void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DO
         return;
     }
 
+    auto source = sourceFromContext(*context);
+    if (!source) {
+        promise.reject(Exception { InvalidStateError, "The context is invalid"_s });
```

Comment:
Instead of InvalidState, this might be an NotSupportedError and "Permissions::query is not supported in this context"